app/migrations.go

```diff
@@ -220,12 +222,12 @@ func (a *App) DoGuestRolesCreationMigration() {
 			} else {
 				scheme.DefaultChannelGuestRole = savedRole.Name
 			}
+		}
 
-			_, err := a.Srv().Store.Scheme().Save(scheme)
-			if err != nil {
-				mlog.Critical("Failed to update custom scheme.", mlog.Err(err))
-				allSucceeded = false
-			}
+		_, err := a.Srv().Store.Scheme().Save(scheme)
```

Comment:
Now you are saving this for every scheme not only for the ones that needs the migration. It should go inside the if block checking for empty DefaultChannelGuestRole or DefaultTeamGuestRole.

Reply:
Oh dear, I've messed up with the brackets. Thanks.

Reply:
yes, gofmt is great until it fixes something that you didn't expect to be fixed :)